File: Elevator.API/Controllers/ElevatorController.cs
using Elevator.API.Application.Interfaces;
using Elevator.API.Domain.Models;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace Elevator.API.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class ElevatorController : ControllerBase
    {
        private readonly IElevatorDoorControl _elevatorDoorControl;
        private readonly IElevatorMovement _elevatorMovement;
        public ElevatorController(IElevatorDoorControl elevatorDoorControl, IElevatorMovement elevatorMovement)
        {
            _elevatorDoorControl = elevatorDoorControl;
            _elevatorMovement = elevatorMovement;
        }

        [HttpPost("open-doors")]
        public IActionResult OpenDoors()
        {
            _elevatorDoorControl.OpenDoors();
            return Ok();
        }

        [HttpPost("close-doors")]
        public IActionResult CloseDoors()
        {
            _elevatorDoorControl.CloseDoors();
            return Ok();
        }

        [HttpPost("call")]
        public IActionResult Call([FromBody] int floor)
        {
            _elevatorMovement.CallElevator(floor);
            return Ok();
        }

        [HttpPost("start")]
        public IActionResult Start()
        {
            _elevatorMovement.Start();
            return Ok();
        }

        [HttpPost("stop")]
        public IActionResult Stop()
        {
            _elevatorMovement.Stop();
            return Ok();
        }

        [HttpGet("status")]
        public ActionResult<ElevatorStatus> GetStatus()
        {
            return _elevatorMovement.GetStatus();
        }
    }
}

File: elevator-frontend/controllers/elevatorController.js
angular.module('elevatorApp')
  .controller('ElevatorController', ['$scope', 'ElevatorService', function($scope, ElevatorService) {
    $scope.floor = 0;
    $scope.status = {};

    $scope.callElevator = function() {
      ElevatorService.call($scope.floor).then(loadStatus);
    };

    $scope.start = function() {
      ElevatorService.start().then(loadStatus);
    };

    $scope.stop = function() {
      ElevatorService.stop().then(loadStatus);
    };

    $scope.openDoors = function() {
      ElevatorService.openDoors().then(loadStatus);
    };

    $scope.closeDoors = function() {
      ElevatorService.closeDoors().then(status);
    };

    function loadStatus() {
      ElevatorService.getStatus().then(function(response) {
        $scope.status = response.data;
      });
    }

    $scope.$on('elevatorStatusUpdated', function (event, status) {
    $scope.status = status;
    $scope.$apply();
});

    loadStatus();
  }]);
